components/dot_menu/dot_menu.tsx

```diff
@@ -112,19 +130,22 @@ export default class DotMenu extends React.PureComponent {
             openUp: false,
             width: 0,
             canEdit: props.canEdit && !props.isReadOnly,
+            canDelete: props.canDelete && !props.isReadOnly,
```

Comment:
I don't think these need to be set since `getDerivedStateFromProps` is called on first mount

Reply:
@hmhealey Are you suggesting to remove this line outright? Removing this line causes the `this.state` assignment to have a type error.

Reply:
Functionally, it could be removed, but it's fine to leave if TS wants it